### ISSUE TITLE
De-serializing Escaped JSON Strings

### ISSUE THREAD

**Issue by jaret32** [OData/AspNetCoreOData#67]
One of my OData entities contains a field that holds HTML content as a string. So when I send a JSON request body, I have to escape double quotes and other special characters. However, OData is not deserializing these escaped JSON strings correctly. I have reproduced the issue in a minimal repository [here](https://github.com/jaret32/ODataWebApi).

Request:
```
{
    "html": "<div class=\"test\"><\/div>"
}
```

Response:
```
{
    "Html": "<div class=\\\"test\\\"></div>"
}
```

**Comment by xuzhg** [comment]
@jaret32  does https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Json/ODataStringEscapeOption.cs#L12 work for you?

**Comment by jaret32** [comment]
@xuzhg I don't believe so as the issue is with the deserialization, not the serialization. How would I go about setting that option so I could try it out?

**Comment by xuzhg** [comment]
You can inject a JsonWriterFactory.

 containerBuilder.AddService(ServiceLifetime.Singleton, sp => new DefaultJsonWriterFactory(stringEscapeOption)

**Comment by jaret32** [comment]
This did not help. I've upgraded to .NET 5.0 and Microsoft.AspNetCore.OData v8.0.0-preview3 and it is working as expected. Except now the fields in the POST request are case sensitive and don't get matched to the model if they are not the same case.

`"html": ...` doesn't work but `"Html": ...` does.

**Comment by xuzhg** [comment]
> This did not help. I've upgraded to .NET 5.0 and Microsoft.AspNetCore.OData v8.0.0-preview3 and it is working as expected. Except now the fields in the POST request are case sensitive and don't get matched to the model if they are not the same case.
> 
> `"html": ...` doesn't work but `"Html": ...` does.

Alright, the payload is case-sensitive.
